Source/WebCore/platform/mediastream/mac/ScreenCaptureKitCaptureSource.mm

```diff
@@ -468,63 +470,11 @@ static void findSharableDevice(RetainPtr<SCShareableContent>&& shareableContent,
             }
         );
 
-        return { static_cast<int>(frame.size.width), static_cast<int>(frame.size.height) };
-    }
-#if HAVE(SC_CONTENT_SHARING_SESSION)
-    IntSize size;
-    BinarySemaphore semaphore;
-    [PAL::getSCShareableContentClass() getShareableContentWithCompletionHandler:makeBlockPtr([&size, &semaphore, deviceID = m_deviceID, deviceType = m_captureDevice.type()] (SCShareableContent *shareableContent, NSError *error) mutable {
-        auto scope = makeScopeExit([&semaphore] {
-            semaphore.signal();
-        });
-        if (error) {
-            RELEASE_LOG_ERROR(WebRTC, "ScreenCaptureKitCaptureSource::intrinsicSize getShareableContentWithCompletionHandler failed with error %s", [[error localizedDescription] UTF8String]);
-            return;
-        }
-        if (deviceType == CaptureDevice::DeviceType::Screen) {
-            [[shareableContent displays] enumerateObjectsUsingBlock:makeBlockPtr([&] (SCDisplay *display, NSUInteger, BOOL *stop) {
-                if (display.displayID == deviceID) {
-                    size = { static_cast<int>(display.width), static_cast<int>(display.height) };
-                    *stop = YES;
-                }
-            }).get()];
-            return;
-        }
-        if (deviceType == CaptureDevice::DeviceType::Window) {
-            [[shareableContent windows] enumerateObjectsUsingBlock:makeBlockPtr([&] (SCWindow *window, NSUInteger, BOOL *stop) {
-                if (window.windowID == deviceID) {
-                    size = { static_cast<int>(window.frame.size.width), static_cast<int>(window.frame.size.height) };
-                    *stop = YES;
-                }
-            }).get()];
-        }
-    }).get()];
-    semaphore.wait();
-    return size;
-#else
-    if (m_captureDevice.type() == CaptureDevice::DeviceType::Screen) {
-        auto displayMode = adoptCF(CGDisplayCopyDisplayMode(m_deviceID));
-        auto screenWidth = CGDisplayModeGetPixelsWide(displayMode.get());
-        auto screenHeight = CGDisplayModeGetPixelsHigh(displayMode.get());
-
-        return { Checked<int>(screenWidth), Checked<int>(screenHeight) };
+        m_intrinsicSize = IntSize(static_cast<int>(frame.size.width), static_cast<int>(frame.size.height));
+        return m_intrinsicSize.value();
     }
 
-    CGRect bounds = CGRectZero;
-    forEachNSWindow([&] (NSDictionary *windowInfo, unsigned windowID, const String&) mutable {
-        if (windowID != m_deviceID)
-            return false;
-
-        NSDictionary *boundsDict = windowInfo[(__bridge NSString *)kCGWindowBounds];
-        if (![boundsDict isKindOfClass:NSDictionary.class])
-            return false;
-
-        CGRectMakeWithDictionaryRepresentation((CFDictionaryRef)boundsDict, &bounds);
-        return true;
-    });
-
-    return { static_cast<int>(bounds.size.width), static_cast<int>(bounds.size.height) };
-#endif
+    return { 640, 480 };
```

Comment:
Shouldn't we return 0,0?

Something like `if (!m_content) return { 0, 0 };`

Reply:
Returning 0x0 regresses the fix for https://bugs.webkit.org/show_bug.cgi?id=247310. We should fix this, but lets do it in another patch.